c-deps/libroach/testutils.cc

```diff
@@ -30,38 +31,49 @@ void __attribute__((weak)) rocksDBLog(char*, int) {}
 
 namespace testutils {
 
-TempDirHandler::TempDirHandler() {}
-
-TempDirHandler::~TempDirHandler() {
-  if (tmp_dir_ == "") {
-    return;
+static int nftw_unlink_cb(const char* name, const struct stat*, int type, struct FTW*) {
+  if (type == FTW_DP) {
+    return rmdir(name);
+  } else if (type == FTW_F || type == FTW_SL) {
+    return unlink(name);
   }
-  std::experimental::filesystem::remove_all(tmp_dir_);
+  return 0;
 }
 
-bool TempDirHandler::Init() {
-  auto fs_dir = std::experimental::filesystem::temp_directory_path() / "tmpccl.XXXXXX";
+TempDirHandler::TempDirHandler() {
+  const char* ostmpdir = getenv("TEMPDIR");
+  if (ostmpdir == NULL) {
+    ostmpdir = "/tmp";
+  }
+  size_t len = strlen(ostmpdir);
+  if (ostmpdir[len - 1] == '/') {
+    len--;
+  }
 
   // mkdtemp needs a []char to modify.
-  char* tmpl = new char[strlen(fs_dir.c_str()) + 1];
-  strncpy(tmpl, fs_dir.c_str(), strlen(fs_dir.c_str()));
+  const char* dirname = "/roachccl.XXXXXX";
+  char* tmpl = new char[len + strlen(dirname) + 1];
+  strncpy(tmpl, ostmpdir, len);
+  strcpy(&tmpl[len], dirname);
 
-  auto tmp_c_dir = mkdtemp(tmpl);
-  if (tmp_c_dir != NULL) {
-    tmp_dir_ = std::string(tmp_c_dir);
-  } else {
-    std::cerr << "Error creating temp directory" << std::endl;
+  if (mkdtemp(tmpl) == NULL) {
+    err(1, "creating temporary directory %s", tmpl);
   }
-
-  delete[] tmpl;
-  return (tmp_c_dir != NULL);
+  tmp_dir_ = tmpl;
```

Comment:
You still need to delete `tmpl`, `tmp_dir_` is a basic_string that will hold a copy of the contents of `tmpl` up to the null character.

Reply:
Oops, thanks. Fixed.